docs/config/README.md

```diff
@@ -448,6 +448,7 @@ The `display` option is an array of the following table.
 | Option               | Default    | Description                                                                                               |
 | -------------------- | ---------- | --------------------------------------------------------------------------------------------------------- |
 | `threshold`          | `10`       | The upper bound for the display option.                                                                   |
+| `only_on_discharge`  | `false`    | Display only when discharging.                                                                            |
```

Comment:
I wonder if we can make this more generic, instead providing a `when` that accepts various values: `charging`, `discharging`, `all`.
This way, folks can change the style depending on status (e.g. green when charging, red when discharging).

Reply:
That would definitely be useful. I'll add some commits to this PR for that, but I'm still unsure about the following points:

- Instead of specifying just a single battery state for the `when` clause, what about implementing it as a list so the same style can be applied to multiple battery states?
- Besides `charging` and `discharging`, the `State` enum also has types like `empty` or `full`. Maybe we could use this enum for the config field? It could be implemented as an `Option` so that a missing field (`None`) would apply the display config to all charging states.

Reply:
I think both ideas sound good. We're slowly moving in this direction in other modules as well (using a list of states instead of 5+ separate binary toggles).

Reply:
Update: I tried to implement the `when` clause as an `Option<Vec<State>>`. I added the `Serialize` and `Deserialize` traits to `State` and although the config schema can be generated, Starship fails to load the config with the error message that `when` is given as a string when it expects an enum.

What I also attempted is to use the `serde_with` crate to serialize/deserialize `State` with the existing `Display` and `Serialize` implementations respectively using `DisplayFromStr`. However, this conflicts with the config schema generation where it cannot serialize `State` as `DisplayFromStr` does not implement `JsonSchema`...

Reply:
Does `#[cfg_attr(feature = "config-schema", derive(schemars::JsonSchema))]` not work?

Reply:
I'll probably have to implement `JsonSchema` for `State`, that's true. Using `serde_as`, I couldn't get the default derive for `JsonSchema` to work as it conflicts with `DisplayFromStr` for not implementing `JsonSchema`.

Reply:
Do `#[serde(deserialize_with = "path")]` and `#[serde(deserialize_with = "path")]` or `#[serde(remote = "...")]` work? If not, I think just adding a `State` `enum` to starship would be best.

Reply:
After trying around with `#[serde(remote = "...")]` and manually implementing the serialization logic for `JsonSchema`, I think the best way to implement the feature is to just add the traits for `serde` and `JsonSchema` directly to `starship_battery::State`. `serde_with` and `JsonSchema` don't seem to play along nicely so as far as I can tell you have to write the serialization logic twice to use them with _remote_ enums.

I'll go ahead and open a MR in the other repo to add the `serde` and `JsonSchema` traits if that's okay.

Reply:
Sounds good.